server/sonar-server/src/main/java/org/sonar/server/test/index/TestIndexer.java

```diff
@@ -60,31 +61,36 @@ public void indexProject(String projectUuid, Cause cause) {
         break;
       case NEW_ANALYSIS:
         deleteProject(projectUuid);
-        super.index(lastUpdatedAt -> doIndex(lastUpdatedAt, projectUuid));
+        doIndex(projectUuid, cause);
         break;
       default:
         // defensive case
         throw new IllegalStateException("Unsupported cause: " + cause);
     }
   }
 
-  public long index(Iterator<FileSourcesUpdaterHelper.Row> dbRows) {
-    BulkIndexer bulk = new BulkIndexer(esClient, INDEX);
-    return doIndex(bulk, dbRows);
+  @Override
+  public Set<IndexTypeId> getIndexTypes() {
+    return ImmutableSet.of(INDEX_TYPE_TEST);
   }
 
   @Override
-  protected long doIndex(long lastUpdatedAt) {
-    return doIndex(lastUpdatedAt, null);
+  public void initializeOnStartup() {
+    doIndex(null, Cause.NEW_ANALYSIS);
+  }
+
+  public long index(Iterator<FileSourcesUpdaterHelper.Row> dbRows) {
+    BulkIndexer bulk = new BulkIndexer(esClient, INDEX_TYPE_TEST.getIndex());
+    return doIndex(bulk, dbRows);
   }
 
-  private long doIndex(long lastUpdatedAt, @Nullable String projectUuid) {
-    final BulkIndexer bulk = new BulkIndexer(esClient, INDEX);
-    bulk.setLarge(lastUpdatedAt == 0L);
+  private long doIndex(@Nullable String projectUuid, Cause cause) {
+    final BulkIndexer bulk = new BulkIndexer(esClient, INDEX_TYPE_TEST.getIndex());
+    bulk.setLarge(cause == Cause.NEW_ANALYSIS);
```

Comment:
Not at all ! Replica must not be disabled each time a new analysis is triggered.